packages/react-dev-overlay/src/internal/container/BuildError.tsx

```diff
@@ -61,4 +67,97 @@ export const styles = css`
   .nextjs-container-build-error-body small {
     color: #757575;
   }
+
+  .nextjs-container-build-error-version-status {
+    font-weight: bold;
+  }
+  .nextjs-container-build-error-version-status span {
+    display: inline-block;
+    width: 10px;
+    height: 10px;
+    border-radius: 5px;
+    background: #eaeaea;
+    margin-right: 6px;
+  }
+
+  .nextjs-container-build-error-version-status span.fresh {
+    background: #50e3c2;
+  }
+
+  .nextjs-container-build-error-version-status span.outdated {
+    background: #e00;
+  }
+
+  .nextjs-container-build-error-version-status span.stale {
+    background: #f5a623;
+  }
 `
+declare global {
+  interface Window {
+    next: { version: string; latest: string }
+  }
+}
+
+type Staleness = 'fresh' | 'stale' | 'outdated'
+
+const NextStatus = () => {
+  const [staleness, setStaleness] = React.useState<Staleness>()
+  React.useEffect(() => {
+    ;(async () => {
+      try {
+        const installedVersion = window.next.version.replace(/-canary.*/, '')
+
+        const latestVersion = await getLatest()
+        console.log(latestVersion, installedVersion)
+
+        const result = compareVersions(installedVersion, latestVersion)
+        setStaleness(result)
+      } catch {}
+    })()
+  }, [])
+
+  const title = {
+    fresh: 'Next.js is up to date!',
+    stale: 'Next.js is out of date, update recommended!',
+    outdated: 'Next.js is out of date, update necessary!',
+    undetermined: 'Could not determine Next.js version status.',
+  }[staleness ?? 'undetermined']
+
+  return (
+    <div title={title} className="nextjs-container-build-error-version-status">
+      <span className={staleness} />
+      Next.js {window.next.version}
+    </div>
+  )
+}
+
+async function getLatest() {
+  return '12.1.6'
+  if (window.next.latest) {
+    return window.next.latest
+  }
+
+  const res = await fetch(
+    'https://api.github.com/repos/vercel/next.js/releases/latest'
+  )
+
+  const latest = await res.json()
+  const { tag_name } = latest
+  const latestVersion = tag_name.replace(/^v/, '')
+  window.next.latest = latestVersion
+  return latestVersion
+}
+
+function compareVersions(a: string, b: string): Staleness {
```

Comment:
Note: we already have semver available here https://github.com/vercel/next.js/blob/canary/packages/next/compiled/semver/package.json

This check should be done on the server and passed to the client. We can also show the out of date message of `next build` and on boot-up for `next dev` then.

Reply:
I was thinking to do this without pulling in `semver`. Although I think `semver.gt()` is not enough as we want to know if it's behind with less than a major to rather show a warning instead of higher urgency to upgrade.